eidpsam/src/main/java/uk/ac/ucl/eidp/auth/jaspic/HttpHeaderAuthModule.java

```diff
@@ -0,0 +1,83 @@
+package uk.ac.ucl.eidp.auth.jaspic;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import javax.security.auth.Subject;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.auth.message.AuthException;
+import javax.security.auth.message.AuthStatus;
+import javax.security.auth.message.MessageInfo;
+import javax.security.auth.message.MessagePolicy;
+import javax.security.auth.message.callback.CallerPrincipalCallback;
+import javax.security.auth.message.callback.GroupPrincipalCallback;
+import javax.security.auth.message.module.ServerAuthModule;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class HttpHeaderAuthModule implements ServerAuthModule {
+
+  private static final Class<?>[] MESSAGE_TYPES = new Class<?>[]{
+    HttpServletRequest.class, HttpServletResponse.class
+  };
+
+//  public static final String USERNAME_HEADER_KEY = "username_header";
```

Comment:
\[Checkstyle\] WARNING: Comment has incorrect indentation level 0, expected is 2, indentation should be the same level as line 27\.